### ISSUE TITLE
When creating nodes definitions using noderange, error prints but definitions are created

### ISSUE THREAD

**Issue by whowutwut** [xcat2/xcat-core#3023]
This is a usability issue.  When an error comes back from `mkdef` that only pertains to a certain node, we should complete the processing and inform the users that the node definitions are created successfully.... 

### Success case 

Creating 16 node definitions
```
[root@fs3 coral]# mkdef sn[01-16] groups="all,coral_sn_regex,service"
16 object definitions have been created or modified.
[root@fs3 coral]# nodels | grep sn | wc -l
16
```

In this case, we report that 16 objects are created or modified.  This is good. .

### Error case 

Remove all the node definitions created above and create 1 definition...
```
[root@fs3 coral]# rmdef sn[01-16]
16 object definitions have been removed.
[root@fs3 coral]# mkdef sn01 groups="all,coral_sn_regex,service"
1 object definitions have been created or modified.
```

Now running the above command to generate 16 results in the following error message: 
```
[root@fs3 coral]# mkdef sn[01-16] groups="all,coral_sn_regex,service"
Error:
A definition for 'sn01' already exists.
Error: To remove the old definition and replace it with
a new definition use the force '-f' option.
Error: To change the existing definition use the 'chdef' command.
Error: One or more errors occured when attempting to create or modify xCAT
object definitions.
```

This makes me think that 0 definitions are created, but infact the ERROR message only applied to the 1st node in the node range.  All other 15 are created. 
```
[root@fs3 coral]#  nodels | grep sn | wc -l
16
```

So ideally we would check the command and not execute anything if we printed that above, but since it's 
 range, and the error could happen at the end, we can't do this unless we have the ability to roll back changes to the DB.  (do we?) 

So at least fix the following.... 
*  Improve the message.  I don't think we need to mention chdef since the user is doing mkdef. 
* reduce the number of messages, there's 3 error messages for 1 node.  So if we have 100 errors, it would print 300 messages...  not that great.  
* change to warning and indicate that nothing will be modified
* Provide the actual count of the definitions modified...

So example: 
```
Warning: A definition for 'sn01' already exists.  No changes will be made.  Run again with -f option to force replace. 
15 object definitions have been created or modified.
```

**Comment by gurevichmark** [comment]
Fixed by PR #3035

**Comment by whowutwut** [comment]
Verified with 2.13.4 5/15 build: `Version 2.13.4 (git commit 9000ae5d1eedc1c412d2a6f35bc0346ef33ad8af, built Mon May 15 09:37:50 EDT 2017)` 

```
[root@fs3 xcat]# mkdef abc[01-16] groups="all,coral_sn_regex,service"
16 object definitions have been created or modified.

[root@fs3 xcat]# rmdef abc[01-16]
16 object definitions have been removed.
[root@fs3 xcat]# mkdef abc01 groups="all,coral_sn_regex,service"
1 object definitions have been created or modified.

[root@fs3 xcat]# mkdef abc[01-16] groups="all,coral_sn_regex,service"
Warning: A definition for 'abc01' already exists. No changes will be made.  Run again with '-f' option to force replace.
15 object definitions have been created or modified.
[root@fs3 xcat]# nodels | grep abc | wc -l
16
```

Looks good, thanks @gurevichmark
